system/Config/DotEnv.php

```diff
@@ -182,6 +182,12 @@ public function normaliseVariable(string $name, string $value = ''): array
 
 		$value = $this->resolveNestedVariables($value);
 
+		// Handle hex2bin prefix
+		if ($name === 'encryption.key' && substr($value, 0, 8) === 'hex2bin:')
```

Comment:
can use strpos instead of substr: 

```suggestion
              if ($name === 'encryption.key' && strpos($value, 'hex2bin:') === 0)
```